### DISCUSSION TITLE
Question: Working with `trex`

### DISCUSSION THREAD

**Discussion by shellscape** [jurassiscripts/velociraptor#78] [Q&A]
Heyo. I'm new to Deno and have been doing homework about equivalents to my favorite and oft used Node features for tooling and DX. I'm really liking what you all have setup with this project. In my readings I also stumbled across [`trex`](https://nicedoc.io/crewdevio/Trex). If you aren't already aware of it, its main usefulness is in managing import maps, which as a first-version Node user, is appealing for the import DX. It also purports to support some assemblance of NPM scripts. However, I find `vr`s approach and feature set more appealing around scripts.

To that end, do the userbase / maintainers have any thoughts on novel approachs for streamlining use of `trex` for its import maps along side `vr` without having to require my peers to add another CLI command to their muscle memory?

**Comment by shellscape** [comment]
@umbopepato since you moved this to a discussion, would be cool if you interacted a bit. comes off a bit odd otherwise.

**Reply by umbopepato** [reply]
@shellscape forgive my late reply, work is keeping me super busy 🙏

**Comment by umbopepato** [comment]
Hey @shellscape, glad you like Velociraptor 😊

Trex is really cool! I understand that using two different CLIs for dependency management and script running may be a bit cumbersome, but it's also true that their configuration syntax and run CLI command are very similar to vr's: in simple cases adding vr to the tooling is only a matter of renaming `run.json` to `scripts.json` and `trex run` to `vr`. Also, [CLI installation helpers](https://github.com/jurassiscripts/velociraptor/issues/63) when implemented will allow vr users to automate the installation of other CLIs used throughout the project (in this sense your collaborators won't have to install the correct version of `trex` by hand and could use vr scripts to manage the main `trex`-related tasks).

In any case, suggestions are welcome!

**Comment by shellscape** [comment]
I completely understand. Thank you for sharing your thoughts. I'll work on implementing your suggestions and will report back.
